quickwit-metastore/src/metastore/single_file_metastore.rs

```diff
@@ -424,7 +424,7 @@ mod tests {
             let index_metadata = IndexMetadata {
                 index_id: index_id.to_string(),
                 index_uri: "ram://indexes/my-index".to_string(),
-                doc_mapper_type: DocMapperType::AllFlatten,
+                doc_mapper: build_doc_mapper("all_flatten", None).unwrap(),
```

Comment:
why not `Box::new(AllFlatten::default())`

Reply:
So my thinking was that since `build_doc_mapper` is exported then it should be the way to construct Doc_Mapper from outside. instead of exposing the concrete types. The mapper modules were declared as private as `mod all_flatten_mapper;`

Reply:
We can make the concrete DocMapper struct public. The module can stay private, but you can do a public reexport.

```
mod all_flatten_mapper;

pub use self::all_flatten_mapper::AllFlattenMapper;
```